app/src/project/core/views.py

```diff
@@ -0,0 +1,164 @@
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.views.decorators.http import require_POST
+from django.views.decorators.csrf import csrf_exempt
+from .prometheus_protobuf import remote_pb2
+
+import bittensor
+import requests
+import structlog
+import snappy
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from project.core.models import Validator
+from .metrics import series_counter, metrics_counter
+
+logger = structlog.getLogger(__name__)
+retries = Retry(
+    total=3,
+    connect=3,
+    read=3,
+    redirect=3,
+    backoff_factor=0.1,
+    status_forcelist=(),
+    raise_on_status=False,
+)
+
+TIMEOUT = 15
+
+session = requests.Session()
+session.mount('http://', HTTPAdapter(max_retries=retries))
+session.mount('https://', HTTPAdapter(max_retries=retries))
+
+
+@csrf_exempt
+@require_POST
+def prometheus_outbound_proxy(request):
+    if not settings.CENTRAL_PROMETHEUS_PROXY_URL:
+        msg = "CENTRAL_PROMETHEUS_PROXY_URL is not configured"
+        logger.error(msg)
+        return HttpResponse(status=500, content=msg.encode())
+    data = request.body
+
+    prometheus_remote_url = urljoin(settings.CENTRAL_PROMETHEUS_PROXY_URL, 'prometheus_inbound_proxy/')
+
+    try:
+        response = session.post(
+            prometheus_remote_url,
+            data=data,
+            headers={
+                'Bittensor-Signature': settings.BITTENSOR_WALLET().hotkey.sign(data).hex(),
+                'Bittensor-Hotkey': settings.BITTENSOR_WALLET().hotkey.ss58_address,
+                **request.headers,
+            },
+            timeout=TIMEOUT,
+        )
+    except requests.exceptions.RequestException as e:
+        logger.info(f"Sending to central prometheus proxy failed: {e}")
+        return HttpResponse(status=500, content=type(e).__name__)
+
+    logger.debug(f"Central prometheus proxy replied with {response.status_code}, {response.content[:200]}")
+    return HttpResponse(
+        status=response.status_code,
+        headers={k: v for k, v in response.headers.items() if k.lower() not in [
+            'connection', 'keep-alive', 'public',
+            'proxy-authenticate', 'transfer-encoding', 'upgrade'
+        ]},
+        content=response.content,
+    )
+
+
+@csrf_exempt
+@require_POST
+def prometheus_inbound_proxy(request):
+    if not settings.UPSTREAM_PROMETHEUS_URL:
+        msg = "UPSTREAM_PROMETHEUS_URL is not configured"
+        logger.error(msg)
+        return HttpResponse(status=500, content=msg.encode())
+
+    data = request.body
+    ss58_address = request.headers.get('Bittensor-Hotkey')
+    signature = request.headers.get('Bittensor-Signature')
+    if not ss58_address or not signature:
+        msg = "Missing required headers."
+        logger.debug(msg)
+        return HttpResponse(status=400, content=msg)
+
+    if ss58_address not in Validator.objects.filter(active=True).values_list('public_key', flat=True):
```

Comment:
```suggestion
    if not Validator.objects.filter(active=True, public_key=ss58_address).exists():
```

Reply:
that's gonna nuke the caches cache with each new request, right?

Reply:
Oh, I missed the use of cacheops.